### ISSUE TITLE
How to map single module and path with same name?

### ISSUE THREAD

**Issue by geirsagberg** [systemjs/systemjs#838]
I am in the process of migrating from RequireJS to SystemJS/jspm, and have a subdirectory named `jquery` with lots of jQuery-plugins:

    content/shared/js/jquery/jquery.validate.js
    content/shared/js/jquery/jquery.multiselect.js
    ...

I would like to map `jquery/*` to `content/shared/js/jquery/*`. However, I also have added `jquery` via jspm. Here is my config.js:

```
System.config({
  baseURL: "/",
  defaultJSExtensions: true,
  paths: {
    "npm:*": "jspm_packages/npm/*",
    "github:*": "jspm_packages/github/*",
    "jquery/*": "content/shared/js/jquery/*"
  },

  map: {
    "jquery": "github:components/jquery@2.1.4",
  }
});
```

My code:

    var $ = require('jquery');
    require('jquery/jquery.multiselect');
    ....

I would expect `jquery/jquery.multiselect` to be mapped to `content/shared/js/jquery/jquery.multiselect.js`, but instead it is mapped to `/jspm_packages/github/components/jquery@2.1.4/jquery.multiselect.js`. 

How can I have both a map `"jquery": "github:components/jquery@2.1.4"` as well as map `jquery/` to a different directory?

**Comment by geirsagberg** [comment]
Also, when I try to map:

```
map: {
  "domready": "subfolder/domready.js"
}
```
it ends up looking for `subfolder/domready` without the .js extension, apparently because of some kind of recursive mapping issues; if I instead do
```
map: {
  "domready": "subfolder/domready2.js"
}
```
it correctly resolves to `subfolder/domready2.js`.

It would probably be easier if there were 2 separate methods to map modules and paths, e.g. something like this:
```
mapPath: {
  "jquery": "content/shared/js/jquery"
},
map: {
  "jquery": "github:components/jquery@2.1.4"
}
```

**Comment by guybedford** [comment]
map applies very early in the normalization pipeline, while paths apply only at the very end. Also `map` is designed to support packages, that is `pkg/main.js` and `pkg` both referring to the same package.

You could use paths to set `jquery`, but that is probably unadvisable in this case.

Instead I'd suggest doing custom paths like `paths: { "jquery/jquery.*": "..." }`.

From your second comment, I can't quite tell the difference between the first two code examples above?

**Comment by guybedford** [comment]
Actually you're probably best off doing separate maps - `map: { 'jquery/jquery.multiselect': '...' }` etc.

**Comment by geirsagberg** [comment]
The difference in the two code examples was that the second one points to `domready2.js` (note the 2) e.g. a file that doesn't exist. However I was unable to reproduce the issue regarding missing extensions with a separate module.

As for separate maps, that's what I was afraid of. I ended up working around the issue by changing the `baseURL` to `content/shared/js` and putting jquery.js in the root of the baseURL, so no `jquery` mapping at all. To keep `jspm_packages` and `config.js` at the project root, I modified the package.json:
```
  "jspm": {
    "directories": {
      "baseURL": "content/shared/js",
      "packages": "jspm_packages"
    },
    "configFile": "config.js"
  }
```
Seems to work out well so far, though I am still a little fuzzy on the difference between `baseURL` in package.json vs config.js. I suppose since the project root is the same as the client URL root, they are the same also?
